util/make_new_dif.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# make_new_dif.py is a script for quickly applying replacement operations to
+# dif_template.h.tpl. See sw/device/lib/dif/dif_template.h.tpl for more
+# information.
+#
+# The produced file may still require spell-checking and general cleaning-up.
+
+import argparse
+
+from pathlib import Path
+
+from mako.template import Template
+
+# This file is $REPO_TOP/util/make_new_dif.py, so it takes two parent()
+# calls to get back to the top.
+REPO_TOP = Path(__file__).resolve().parent.parent
+
+
+def main():
+    dif_dir = REPO_TOP / 'sw/device/lib/dif'
+    template_file = dif_dir / 'dif_template.h.tpl'
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--ip',
+                        '-i',
+                        required=True,
+                        help='the short name of the IP, in snake_case')
+    parser.add_argument('--peripheral',
+                        '-p',
+                        required=True,
+                        help='the documentation-friendly name of the IP')
+    parser.add_argument(
+        '--handle_param',
```

Comment:
nit: arguments are generally with `-` in OT scripts.

Reply:
That doesn't seem to be the case from looking at a couple of random scripts:
https://github.com/lowRISC/opentitan/blob/master/util/lint_commits.py
https://github.com/lowRISC/opentitan/blob/master/util/build_docs.py
https://github.com/lowRISC/opentitan/blob/master/util/vendor.py

Maybe I'm missing something. Either way, two dashes seems consistent with what the `argparse` documentation suggests: https://docs.python.org/3/library/argparse.html

Reply:
I meant within the argument, i.e. "--handle-param" in this case.

Reply:
Oh yup that makes sense.